{
  "additionalMetadata": {
    "cna": "mitre",
    "cveId": "CVE-2023-38898",
    "description": "An issue in Python cpython v.3.7 allows an attacker to obtain sensitive information via the _asyncio._swap_current_task component. NOTE: this is disputed by the vendor because (1) neither 3.7 nor any other release is affected (it is a bug in some 3.12 pre-releases); (2) there are no common scenarios in which an adversary can call _asyncio._swap_current_task but does not already have the ability to call arbitrary functions; and (3) there are no common scenarios in which sensitive information, which is not already accessible to an adversary, becomes accessible through this bug.",
    "reason": "Improve version ranges to indicate fix",
    "references": [
      "https://github.com/python/cpython/issues/105987"
    ]
  },
  "adp": {
    "affected": [
      {
        "collectionURL": "https://github.com",
        "cpes": [
          "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
        ],
        "packageName": "python/cpython",
        "product": "CPython",
        "repo": "https://github.com/python/cpython",
        "vendor": "Python Software Foundation",
        "versions": [
          {
            "lessThan": "3.13.0a1",
            "status": "affected",
            "version": "3.13.0a0",
            "versionType": "python"
          },
          {
            "lessThan": "3.12.0b4",
            "status": "affected",
            "version": "0",
            "versionType": "python"
          }
        ]
      }
    ],
    "providerMetadata": {
      "orgId": "00000000-0000-4000-8000-000000000000",
      "shortName": "anchoreadp"
    },
    "references": [
      {
        "url": "https://github.com/python/cpython/pull/106099"
      },
      {
        "url": "https://github.com/python/cpython/pull/105989"
      }
    ]
  }
}